src/extension/wps/wps-core/src/test/java/org/geoserver/wps/ExecuteTest.java

```diff
@@ -751,6 +755,43 @@ public void testFeatureCollectionFileReferenceKVPError() throws Exception {
                 "/wps:ExecuteResponse/wps:Status/wps:ProcessFailed/ows:ExceptionReport/ows:Exception/ows:ExceptionText",
                 d);
     }
+    
+    @Test
+    public void testHttpSecutiry() throws Exception {
+        URLEntry googleOnly =
+                new URLEntry(
+                        "google only",
+                        "only allow url starting with http://www.google.com",
+                        "^(http://www.google).*$");
+        addURLEntryGeoserverURLConfigService(googleOnly);
+        //GeoserverURLConfigService.getSingleton().addAndsave(googleOnly);
+        //tests static methods in  org.vfny.geoserver.util.Requests
+        enableGeoserverURLConfigService(true);
+        //WPS process should  through exception when evaluating this resource
+        String externalNotAllowedURL = "http://www.unallowed.com/resource.xml";
+        String request =
+                "wps?service=WPS&version=1.0.0&request=Execute&Identifier=gs:BufferFeatureCollection"
+                        + "&DataInputs="
+                        + urlEncode(
+                                "features=@mimetype=application/wfs-collection-1.1@xlink:href="
+                                        + externalNotAllowedURL
+                                        + ";distance=10")
+                        + "&ResponseDocument="
+                        + urlEncode("result");
+        
+        Document d = getAsDOM(request);
+        // print(d);
+        checkValidationErrors(d);
+        
+        assertEquals("wps:ExecuteResponse", d.getDocumentElement().getNodeName());
+        
+        assertXpathExists("/wps:ExecuteResponse/wps:Status/wps:ProcessFailed", d);
+        assertXpathEvaluatesTo(
+                "Failed to retrieve value for input features\n" + 
+                "&gt;&gt;&gt;Evaluation Failure: http://www.unallowed.com/resource.xml: did not pass security evaluation",
```

Comment:
I would say something like "Connections to http://www.unallowed.com/resource.xml are not allowed by the security configuration"

Reply:
this error comes from Geotools

Reply:
Then fix it in GeoTools, it's still part of the HTTP authorization work no?